### PR TITLE
[#59] 판매 상품 재고 조회 기능

### PR DESCRIPTION
**Pull request by JinDDung2**
## 작업 요약

- 판매 재고 조회
- 판매 재고 세부 조회

## 작업 내용

> (체크 박스를 모두 선택해야 merge가 활성화 됩니다.)

- [x]  재고 조회할 때 미구매상품 + 만료기간 가까운 순
- [x]  재고 세부 조회할 때 미구매상푸만 조회

### CODE REVIEW

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [97.31%]|:green_apple:|
|:-|:-:|:-:|
|[Sale.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fdomain%2FSale.java)|100%|:green_apple:|
|[ErrorCode.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Fexception%2FErrorCode.java)|100%|:green_apple:|
|[SaleItemFacade.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleItemFacade.java)|100%|:green_apple:|
|[SaleCreationFacade.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleCreationFacade.java)|100%|:green_apple:|
|[SaleController.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fcontroller%2FSaleController.java)|100%|:green_apple:|
|[SaleService.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fservice%2FSaleService.java)|97.18%|:green_apple:|
|[AuthInterceptor.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Fsecurity%2Finterceptor%2FAuthInterceptor.java)|96.03%|:green_apple:|
|[SaleCreateValidator.java](https://github.com/f-lab-edu/givemeticon/blob/c64219edfde2f4f23e24bedc29e66f96c1cfff08/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fvalidator%2FSaleCreateValidator.java)|62.5%|:x:|

|Total Project Coverage|90.85%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/jinddung2/givemeticon/domain/sale/controller/request/SaleCreateRequest.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public Sale toEntity() {
         return Sale.builder()
                 .barcode(barcode)
                 .expirationDate(expirationDate)
-                .isUsed(false)
+                .isBought(false)
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/sale/controller/request/SaleCreateRequest.java:22]: 👍

File: src/main/java/com/jinddung2/givemeticon/domain/sale/facade/SaleItemFacade.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class SaleItemFacade {
+
+    private final ItemService itemService;
+    private final SaleService saleService;
+
+    public List<SaleDto> getSalesByItemId(int itemId) {
+        itemService.getItem(itemId);
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/sale/facade/SaleItemFacade.java:19]: 왜 가져와서 아무것도 안하지? 하고 봤더니 안에서 여러가지 일을 하고 있군요?
https://github.com/f-lab-edu/givemeticon/blob/develop/src/main/java/com/jinddung2/givemeticon/domain/item/service/ItemService.java#L22-L27

기능 단위로 리팩토링이 필요해 보입니다~

File: src/test/java/com/jinddung2/givemeticon/domain/sale/controller/SaleControllerTest.java [hunk 3]
```diff
@@ -145,4 +157,63 @@ void create_Sale_Fail_DUPLICATED_BARCODE_NUMBER() throws Exception {
                 .andExpect(jsonPath("$.message").value("FAIL"))
                 .andExpect(jsonPath("$.data.message").value("이미 등록된 바코드 입니다."));
     }
+
+    @Test
```
Comment (f-lab-kyle) [inline src/test/java/com/jinddung2/givemeticon/domain/sale/controller/SaleControllerTest.java:167]: 따로 언급은 안드렸었는데, 테스트 코드 작성할때 한글로 상황과 기대값을 표현하는게 더 가독성에 좋지 않을까 생각이 들어요. 그렇다면 @DisplayName 을 사용해도 좋을것 같고요 ~

영어로 충분히 이해가 된다면 패스 하셔도 좋습니다~

File: src/main/java/com/jinddung2/givemeticon/domain/sale/facade/SaleItemFacade.java (hunk 2 above)
Comment (JinDDung2) [reply src/main/java/com/jinddung2/givemeticon/domain/sale/facade/SaleItemFacade.java:19]: 현재 ItemService.getItem()은 아이템 정보도 가져오고, 조회수 증가 역할도 하게 되어있습니다.
제가 생각한 방안은 다음과 같습니다.
1. ItemService.validateItem() -> 아이템을 가져오고, 없다면 예외처리하기
2. ItemService.getItem() -> validateItem 메서드 호출 후 조회수 증가
하지만 위와 같이 하더라도 예외처리는 ItemService에서 진행되니 이 클래스의 getSalesByItemId() 메서드는 validateItem 메서드 실행 후 추가 작업이 필요해 보이진 않습니다. 다른 방안이 있을까요?

File: src/test/java/com/jinddung2/givemeticon/domain/sale/controller/SaleControllerTest.java (hunk 3 above)
Comment (JinDDung2) [reply src/test/java/com/jinddung2/givemeticon/domain/sale/controller/SaleControllerTest.java:167]: 생각해보니 서비스 테스트에만 @DisplayName을 활용해서 설명한 것 같습니다. 컨트롤러 테스트도 한글로, 서술형으로 테스트 목적을 설명하겠습니다.

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [97.82%]|:green_apple:|
|:-|:-:|:-:|
|[Sale.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fdomain%2FSale.java)|100%|:green_apple:|
|[ErrorCode.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Fexception%2FErrorCode.java)|100%|:green_apple:|
|[BrandService.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fbrand%2Fservice%2FBrandService.java)|100%|:green_apple:|
|[SaleItemFacade.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleItemFacade.java)|100%|:green_apple:|
|[SaleCreationFacade.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleCreationFacade.java)|100%|:green_apple:|
|[SaleController.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fcontroller%2FSaleController.java)|100%|:green_apple:|
|[SaleService.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fservice%2FSaleService.java)|97.18%|:green_apple:|
|[AuthInterceptor.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Fsecurity%2Finterceptor%2FAuthInterceptor.java)|96.03%|:green_apple:|
|[SaleCreateValidator.java](https://github.com/f-lab-edu/givemeticon/blob/e1c025dff600847994ecdc8d8cd34bcce97d2573/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fvalidator%2FSaleCreateValidator.java)|62.5%|:x:|

|Total Project Coverage|91.83%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**